sycl/source/detail/scheduler/commands.hpp

```diff
@@ -698,6 +716,17 @@ class KernelFusionCommand : public Command {
   FusionStatus MStatus;
 };
 
+// Enqueues a given kernel to a RT::PiExtCommandBuffer
+pi_int32 enqueueImpCommandBufferKernel(
```

Comment:
Would it be possible to reduce the number of parameters here by passing a `CGExecKernel`, which encapsulates most of the parameters used here?

Reply:
> Do you only go through the scheduler for the recording mode or also for the explicit API?

Since it happens at finalize time both woudl eventually end up going through the scheduler.

> The extension proposal does not allow simultaneous use of fusion and graph recording Do you already emit any diagnostic/error in that case? If not, would the scheduler be a good place for that?

We don't currently emit one in the implementation but it is in the spec. The error would be emitted earlier though when starting recording or fusion, rather than at finalize time in the scheduler.